astropy/coordinates/transformations/affine.py

```diff
@@ -0,0 +1,361 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+
+"""
+This module contains a general framework for defining graphs of transformations
+between coordinates, suitable for either spatial coordinates or more generalized
+coordinate systems.
+
+The fundamental idea is that each class is a node in the transformation graph,
+and transitions from one node to another are defined as functions (or methods)
+wrapped in transformation objects.
+
+This module also includes more specific transformation classes for
+celestial/spatial coordinate frames, generally focused around matrix-style
+transformations that are typically how the algorithms are defined.
+"""
+
+
+from abc import abstractmethod
+
+import numpy as np
+
+from astropy.coordinates.transformations.base import CoordinateTransform
+
+__all__ = []
+
+
+class BaseAffineTransform(CoordinateTransform):
+    """Base class for common functionality between the ``AffineTransform``-type
+    subclasses.
+
+    This base class is needed because `~astropy.coordinates.AffineTransform`
+    and the matrix transform classes share the ``__call__()`` method, but
+    differ in how they generate the affine parameters.
+    `~astropy.coordinates.StaticMatrixTransform` passes in a matrix stored as a
+    class attribute, and both of the matrix transforms pass in ``None`` for the
+    offset. Hence, user subclasses would likely want to subclass this (rather
+    than `~astropy.coordinates.AffineTransform`) if they want to provide
+    alternative transformations using this machinery.
+
+    """
+
+    def _apply_transform(self, fromcoord, matrix, offset):
+        from astropy.coordinates.representation import (
```

Comment:
I'm surprised there would be an import problem here, since `representation` does not need any of the transformation stuff. Though for follow-up, best not to change code things here!!